pkg/xds/envoy/metadata/v3/metadata.go

```diff
@@ -33,22 +32,18 @@ func LbMetadata(tags envoy_common.Tags) *envoy_core.Metadata {
 		return nil
 	}
 	fields := MetadataFields(tags)
-	metadata := &envoy_core.Metadata{
+	return &envoy_core.Metadata{
 		FilterMetadata: map[string]*structpb.Struct{
 			"envoy.lb": {
 				Fields: fields,
 			},
 		},
 	}
-	return metadata
 }
 
 func MetadataFields(tags envoy_common.Tags) map[string]*structpb.Value {
 	fields := map[string]*structpb.Value{}
 	for key, value := range tags {
-		if key == mesh_proto.ServiceTag {
```

Comment:
Why is this and the changes to the TLS configurers necessary?

Reply:
This method was previously building metadata without kuma.io/service tag. I need kuma.io/service tag for listener matcher. I now exclude this tag for those 2 usages of the method outside of listener matcher.